src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1087,34 +1091,46 @@ internal static string GetExpandedEnvironmentVariable(string name, EnvironmentVa
+        private static bool PathContainsSubstring(string pathToScan, string pathToLookFor)
         {
-            // we don't support if any of the args are null - parent function should ensure this; empty values are ok
-            Diagnostics.Assert(pathToScan != null, "pathToScan should not be null according to contract of the function");
-            Diagnostics.Assert(pathToLookFor != null, "pathToLookFor should not be null according to contract of the function");
+            // We don't support if any of the args are null - parent function should ensure this; empty values are OK.
+            Dbg.Assert(pathToScan != null, "pathToScan should not be null according to contract of the function");
+            Dbg.Assert(pathToLookFor != null, "pathToLookFor should not be null according to contract of the function");
+
+            if (pathToLookFor.Length is 0)
+            {
+                // An empty path to look for is always considered found in 'pathToScan'.
+                return true;
+            }
+
+            if (pathToScan.Length is 0)
+            {
+                // The 'pathToScan' is empty, while 'pathToLookFor' is not - so it can not be found.
+                return false;
+            }
 
-            int pos = 0; // position of the current substring in pathToScan
-            string[] substrings = pathToScan.Split(Path.PathSeparator, StringSplitOptions.None); // we want to process empty entries
-            string goodPathToLookFor = pathToLookFor.Trim().TrimEnd(Path.DirectorySeparatorChar); // trailing backslashes and white-spaces will mess up equality comparison
+            string[] substrings = pathToScan.Split(
+                Path.PathSeparator,
+                StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+
+            // Remove the trailing directory separators and white spaces.
+            string goodPathToLookFor = pathToLookFor.Trim().TrimEnd(Path.DirectorySeparatorChar);
             foreach (string substring in substrings)
             {
-                string goodSubstring = substring.Trim().TrimEnd(Path.DirectorySeparatorChar);  // trailing backslashes and white-spaces will mess up equality comparison
+                // Remove the trailing directory separators. Trailing white spaces were already removed by 'StringSplitOptions.TrimEntries'.
+                string goodSubstring = substring.TrimEnd(Path.DirectorySeparatorChar);
 
                 // We have to use equality comparison on individual substrings (as opposed to simple 'string.IndexOf' or 'string.Contains')
-                // because of cases like { pathToScan="C:\Temp\MyDir\MyModuleDir", pathToLookFor="C:\Temp" }
-
+                // because of cases like { pathToScan="C:\Temp\MyDir\MyModuleDir", pathToLookFor="C:\Temp" }.
                 if (string.Equals(goodSubstring, goodPathToLookFor, StringComparison.OrdinalIgnoreCase))
```

Comment:
Should it be Ordinal on Unix?

Reply:
It's hard to imagine a user would deliberately add a path that only differs from an existing path in letter case. I will keep this unchanged. If we ever need to change this, it should be a separate PR.